fuel-txpool/src/subscribers.rs

```diff
@@ -0,0 +1,49 @@
+use std::sync::Arc;
+
+use crate::{types::ArcTx, Error};
+use async_trait::async_trait;
+use fuel_core_interfaces::txpool::Subscriber;
+use parking_lot::RwLock;
```

Comment:
[nit] Should we be using an async friendly RwLock instead?

Reply:
For subscribers, parking_lot is better fit it is faster in this use case, bcs only `read` lock is used and it is not blocking.

Reply:
Ah right, and we'd only hold the lock when adding subscribers during initialization.